src/RevOps.js

```diff
@@ -0,0 +1,154 @@
+    createAccount: PropTypes.bool,
+
+    /** react components nested inside of this component */
+    children: PropTypes.element,
+
+    /** getToken (accountId) => { access_token } callback function that is called before every call requiring authorization */
+    getToken: PropTypes.func,
+  }
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      email: props.email,
+      account: props.account,
+    }
+  }
+
+  async componentDidMount() {
+    const { createAccount, account, getToken, publicKey } = this.props
+    // when creating, wait to get token until we have an account
+    if (createAccount !== true) {
+      if (!!getToken !== false && typeof (getToken) === 'function') {
+        getToken(account.accountId)
+          .then(accessToken => this.setState({ accessToken }))
+          .catch(error => console.error(error))
+      }
+      this.setState({ account: makeAccount({ ...account }) })
+    }
+    // create an account automatically create an account if we have enough information
+    // if (createAccount === true && !!account.email === true) {
+    //   this.createAccount()
+    // }
+    if (createAccount === true && publicKey === false) {
+      console.warn("Your public key is required for creating accounts.")
+    }
+  }
+
+  // componentDidUpdate(prevProps) {
+  //   const { account } = this.props
+  //   if (prevProps.account !== account) {
+  //     this.setState({ account: makeAccount({ ...account }) })
+  //   }
+  // }
+
+  // Use the public key to create the account
+  // then use the account returned to call getToken and send the token info down stream
+  createAccount = async () => {
+    const { publicKey, account, apiOptions } = this.props
+    const accountCreationURL = `https://vault.revops.io/v1/accounts`
+
+    let options = {
+      method: 'POST',
+      headers: {
+        'Authorization': `Bearer ${publicKey}`,
+        'Content-Type': 'application/json;charset=UTF-8',
+        'X-RevOps-Sandbox': 'true',
+      },
+      body: JSON.stringify({ ...this.state.account, email: this.state.email, account_id: account.accountId })
+    };
+
+    let response = await fetch(accountCreationURL, options)
+    let responseOK = response && response.ok
+    if (responseOK) {
+      let data = await response.json()
+      this.setState({
+        account: makeAccount({ ...data }),
+        loading: false,
+        error: false,
+      })
+      if (!!apiOptions.getToken !== false && typeof (apiOptions.getToken) === 'function') {
+        apiOptions.getToken(data.account_id)
+          .then(accessToken => this.setState({ accessToken }))
+          .catch(error => console.error('Unable to get token', error))
+      }
+    } else {
+      console.log('Unable to create account account')
+      this.setState({
+        account: false,
+        loading: false,
+        error: {} // TODO: Match error format
+      })
+    }
+  }
```

Comment:
Why is this hitting the vault? We should be using VGSProxy. @john-revops I believe this is insecure.